service/gcs/bridge/bridge.go

```diff
@@ -317,7 +340,55 @@ func (b *Bridge) PublishNotification(n *prot.ContainerNotification) {
 	b.responseChan <- resp
 }
 
+// negotiateProtocol was introduced in v4 so will not be called
+// with a minimum lower than that.
+func (b *Bridge) negotiateProtocol(w ResponseWriter, r *Request) {
+	if r.Version != prot.PvInvalid {
+		logrus.Errorf("bridge: Message Type: %v was called more than once.", r.Header.Type)
+		w.Error("", gcserr.NewHresultError(gcserr.HrVmcomputeProtocolError))
+		return
+	}
+
+	var request prot.NegotiateProtocol
+	if err := commonutils.UnmarshalJSONWithHresult(r.Message, &request); err != nil {
+		w.Error("", errors.Wrapf(err, "failed to unmarshal JSON in message \"%s\"", r.Message))
+		return
+	}
+
+	if request.MaximumVersion.Major < uint32(prot.PvV4) || uint32(prot.PvMax) < request.MinimumVersion.Major {
+		w.Error(request.ActivityID, gcserr.NewHresultError(gcserr.HrVmcomputeUnsupportedProtocolVersion))
+		return
+	}
+
+	min := func(x, y uint32) uint32 {
```

Comment:
Could you use `math.Min` instead?

Reply:
Actually you cant. Golang only has math.Min for floats...... And this is the recommend pattern in Go

Reply:
Ah, right.